packages/x-charts-pro/src/context/ZoomProvider/useSetupZoom.ts

```diff
@@ -0,0 +1,272 @@
+        );
+
+        // If the scale ratio is 0, it means the pinch gesture is not valid.
+        if (scaleRatio === 0) {
+          eventPrevDiff.current = curDiff;
+          return;
+        }
+
+        const point = getSVGPoint(element, firstEvent);
+
+        const centerRatio = getHorizontalCenterRatio(point, area);
+
+        const [newMinRange, newMaxRange] = zoomAtPoint(centerRatio, scaleRatio, zoomRange);
+
+        // TODO: make span a config option.
+        if (!isSpanValid(newMinRange, newMaxRange, isZoomIn)) {
+          eventPrevDiff.current = curDiff;
+          return;
+        }
+
+        eventPrevDiff.current = curDiff;
+        setZoomRange([newMinRange, newMaxRange]);
+      }
+    }
+
+    function pointerUpHandler(event: PointerEvent) {
+      eventCacheRef.current.splice(
+        eventCacheRef.current.findIndex((e) => e.pointerId === event.pointerId),
+        1,
+      );
+
+      if (eventCacheRef.current.length < 2) {
+        eventPrevDiff.current = 0;
+      }
+    }
+
+    element.addEventListener('wheel', wheelHandler);
+    element.addEventListener('pointerdown', pointerDownHandler);
+    element.addEventListener('pointermove', pointerMoveHandler);
+    element.addEventListener('pointerup', pointerUpHandler);
+    element.addEventListener('pointercancel', pointerUpHandler);
+    element.addEventListener('pointerout', pointerUpHandler);
+    element.addEventListener('pointerleave', pointerUpHandler);
+
+    // Prevent zooming the entire page on touch devices
+    element.addEventListener('touchstart', preventDefault);
+    element.addEventListener('touchmove', preventDefault);
+
+    return () => {
+      element.removeEventListener('wheel', wheelHandler);
+      element.removeEventListener('pointerdown', pointerDownHandler);
+      element.removeEventListener('pointermove', pointerMoveHandler);
+      element.removeEventListener('pointerup', pointerUpHandler);
+      element.removeEventListener('pointercancel', pointerUpHandler);
+      element.removeEventListener('pointerout', pointerUpHandler);
+      element.removeEventListener('pointerleave', pointerUpHandler);
+      element.removeEventListener('touchstart', preventDefault);
+      element.removeEventListener('touchmove', preventDefault);
+    };
+  }, [svgRef, setZoomRange, zoomRange, area]);
+};
+
+/**
+ * Checks if the new span is valid.
+ */
+function isSpanValid(minRange: number, maxRange: number, isZoomIn: boolean) {
+  const newSpanPercent = maxRange - minRange;
+
+  // TODO: make span a config option.
+  if (
+    (isZoomIn && newSpanPercent < MIN_ALLOWED_SPAN) ||
+    (!isZoomIn && newSpanPercent > MAX_ALLOWED_SPAN)
+  ) {
+    return false;
+  }
+
+  return true;
+}
+
+/**
+ * Get the scale ratio and if it's a zoom in or out from a wheel event.
+ */
+function getWheelScaleRatio(event: WheelEvent, step: number) {
+  const deltaY = event.deltaY;
+  const multiplier = isTrackPad(event) ? 1 : 10;
+  const scaledStep = (step * multiplier) / 1000;
```

Comment:
Thaught that could be helpfull here

https://d3js.org/d3-zoom#zoom_wheelDelta

```js
function wheelDelta(event) {
  return -event.deltaY * (event.deltaMode === 1 ? 0.05 : event.deltaMode ? 1 : 0.002) * (event.ctrlKey ? 10 : 1);
}
```

Reply:
Had to adapt it a bit, but seems to work nicely, thanks